src/content/docs/en/reference/dev-overlay-plugin-reference.mdx

```diff
@@ -0,0 +1,313 @@
+
+## Client-side Events
+
+Using the [`eventTarget`](#eventtarget) argument on the [`init`](#init-canvas-shadowroot-eventtarget-eventtarget--void) hook, plugins can send and receive events from the dev overlay. The following events are available:
+
+### `plugin-toggle`
+
+This event is fired when the user clicks on the plugin icon in the dev overlay bar.
+
+```ts title="src/my-plugin.js" {3-6}
+  export default {
+    // ...
+    init(canvas, eventTarget) {
+      eventTarget.addEventListener('plugin-toggle', (event) => {
+        if (event.detail.state === true) {
+          console.log("The plugin is now enabled!")
+        }
+      })
+    }
+  }
+```
+
+#### `state: boolean`
+
+Indicates whether or not the plugin is enabled after the user's click.
+
+### `plugin-notification`
+
+This event can be sent to inform the user that the plugin requires attention.
+
+```ts title="src/my-plugin.js" {3-6}
+  export default {
+    // ...
+    init(canvas, eventTarget) {
+      eventTarget.dispatchEvent(
+          new CustomEvent('plugin-notification', {
+            detail: {
+              state: true,
+            },
+          })
+        );
+    }
+  }
+```
+
+#### `state: boolean`
+
+Indicates whether or not the plugin has a notification for the user. When `true`, the plugin icon will be highlighted using a red dot. Conversely, when `false`, the highlight will be removed. If this property is not specified, `true` will be assumed.
+
+## Client-Server Communication
+
+Using [Vite's methods for client-server communication](https://vitejs.dev/guide/api-plugin.html#client-server-communication), dev overlay plugins can communicate with the server.
+
+In addition to being able to send and receive custom messages, the dev overlay also sends the following messages, where `PLUGIN_ID` is the [plugin's ID](#id-string):
+
+### `astro-dev-overlay:PLUGIN_ID:init`
+
+This message is sent when the plugin is initialized. The data for this message is empty.
+
+```ts title="integration.ts"
+{
+  // ...
+  "astro:server:setup": ({ server }) => {
+    server.ws.on("astro-dev-overlay:super-plugin:init", () => {
+      console.log("My plugin was initialized!");
+    });
+  },
+  // ...
+}
+```
+
+### `astro-dev-overlay:PLUGIN_ID:toggle`
+
+This message is sent when the user clicks on the plugin icon in the UI. The data for this message is a boolean indicating whether the plugin is enabled or not.
+
+```ts title="integration.ts"
+{
+  // ...
+  "astro:server:setup": ({ server }) => {
+    server.ws.on("astro-dev-overlay:super-plugin:toggle", (data) => {
+      console.log(`My plugin is now ${data.state ? "enabled" : "disabled"}!`);
+    });
+  },
+  // ...
+}
+```
+
+:::note
+The built-in `connection` event from Vite fire *before* dev overlay plugins are initialized. As such, trying to receive events from this event will not work inside plugins. Instead, use the `astro-dev-overlay:PLUGIN_ID:init` event.
```

Comment:
```suggestion
The built-in `connection` event from Vite fires **before** dev overlay plugins are initialized and therefore cannot be used directly by plugins. Instead, use the `astro-dev-overlay:PLUGIN_ID:init` event.
```
I know this is a little more generic, but would this be sufficient here?

Reply:
That's perfectly fine with me!